src/coreclr/jit/llvmlower.cpp

```diff
@@ -929,40 +895,51 @@ void Llvm::lowerCallToShadowStack(GenTreeCall* callNode)
 
             if (corInfoType == CORINFO_TYPE_VALUECLASS)
             {
-                shadowStackUseOffest = padNextOffset(corInfoType, clsHnd, shadowStackUseOffest);
+                shadowStackUseOffset = padNextOffset(corInfoType, clsHnd, shadowStackUseOffset);
             }
             else
             {
-                shadowStackUseOffest += TARGET_POINTER_SIZE;
+                shadowStackUseOffset += TARGET_POINTER_SIZE;
             }
+
+            callNode->gtArgs.RemoveAfter(lastLlvmStackArg);
         }
         else
         {
             // Arg on LLVM stack.
             GenTreePutArgType* putArg = _compiler->gtNewPutArgType(argNode, corInfoType, clsHnd);
+            callArg.SetEarlyNode(putArg);
 #if DEBUG
-            putArg->SetArgNum(opAndArg.argNum);
+            putArg->SetArgNum(argIx);
 #endif
-            if (lastArg == nullptr)
-            {
-                lastArg = _compiler->gtNewCallArgs(putArg);
-                callNode->gtCallArgs = lastArg;
-            }
-            else
-            {
-                lastArg = _compiler->gtInsertNewCallArgAfter(putArg, lastArg);
-            }
-
             CurrentRange().InsertBefore(callNode, putArg);
-        }
-
-        if (isSigArg && (sigInfo != nullptr))
-        {
-            sigArgs = _info.compCompHnd->getArgNext(sigArgs);
+            lastLlvmStackArg = &callArg;
         }
 
         argIx++;
     }
+
+    callNode->gtArgs.MoveLateToEarly();
```

Comment:
Can we do this before at the top, before rewriting the argument list? It'd be a bit clearer I think.

Reply:
Yes, done